neqo-transport/src/flow_mgr.rs

```diff
@@ -153,173 +158,157 @@ impl FlowMgr {
+                if self.conn_credit_avail() == 0 {
+                    self.data_blocked()
                 }
             }
             Frame::StreamDataBlocked { stream_id, .. } => {
-                if let Some(ss) = conn.send_streams.get(&stream_id.into()) {
+                if let Ok(ss) = send_streams.get(stream_id.into()) {
                     if ss.credit_avail() == 0 {
-                        conn.flow_mgr
-                            .borrow_mut()
-                            .stream_data_blocked(stream_id.into(), ss.max_stream_data())
+                        self.stream_data_blocked(stream_id.into(), ss.max_stream_data())
                     }
                 }
             }
             Frame::StreamsBlocked { stream_type, .. } => match stream_type {
                 StreamType::UniDi => {
-                    if conn.indexes.peer_next_stream_uni >= conn.indexes.peer_max_stream_uni {
-                        conn.flow_mgr
-                            .borrow_mut()
-                            .streams_blocked(conn.indexes.peer_max_stream_uni, StreamType::UniDi);
+                    if indexes.peer_next_stream_uni >= indexes.peer_max_stream_uni {
+                        self.streams_blocked(indexes.peer_max_stream_uni, StreamType::UniDi);
                     }
                 }
                 StreamType::BiDi => {
-                    if conn.indexes.peer_next_stream_bidi >= conn.indexes.peer_max_stream_bidi {
-                        conn.flow_mgr
-                            .borrow_mut()
-                            .streams_blocked(conn.indexes.peer_max_stream_bidi, StreamType::BiDi);
+                    if indexes.peer_next_stream_bidi >= indexes.peer_max_stream_bidi {
+                        self.streams_blocked(indexes.peer_max_stream_bidi, StreamType::BiDi);
                     }
                 }
             },
             // Resend StopSending
             Frame::StopSending {
                 stream_id,
                 application_error_code,
-            } => conn
-                .flow_mgr
-                .borrow_mut()
-                .stop_sending(stream_id.into(), application_error_code),
+            } => self.stop_sending(stream_id.into(), application_error_code),
             // Resend MaxStreamData if not SizeKnown
             // (maybe_send_flowc_update() checks this.)
             Frame::MaxStreamData { stream_id, .. } => {
-                if let Some(rs) = conn.recv_streams.get_mut(&stream_id.into()) {
+                if let Some(rs) = recv_streams.get_mut(&stream_id.into()) {
                     rs.maybe_send_flowc_update()
                 }
             }
             Frame::PathResponse { .. } => qinfo!("Path Response lost, not re-sent"),
-            _ => qwarn!("Unexpected Flow frame {:?} lost, not re-sent", self.0),
+            _ => qwarn!("Unexpected Flow frame {:?} lost, not re-sent", token),
         }
     }
 }
+
+impl Iterator for FlowMgr {
```

Comment:
Why does this need to be an iterator as opposed to just `get_next_frame` ?

Reply:
lol, I think you made it an Iterator! I don't care :)